core/set.rbs

```diff
@@ -180,7 +180,7 @@
 #      Resets the internal state; useful if an object
 #      has been modified while an element in the set.
 #
-class Set[unchecked out A]
+class Set[unchecked out A < Object]
```

Comment:
One of us should rather send a PR to support `BasicObject`.
Else it’d be a sign that `Set` isn’t stable enough for the highly-requested Set literal.

Reply:
Are you suggesting that we should make modifications to support `BasicObject` in ruby/set as much as possible?
I do not have the motivation to do so myself, but it would certainly be a contribution to the community.

Reply:
FWIW… neither does `Array#inspect`
```ruby
[BasicObject.new].inspect #=! undefined method `inspect' for an instance of BasicObject (NoMethodError)
```
For this situation, IIRC @sampersand had the idea of adding a RBS syntax for additional fine-grained bounds on the method level, e.g.,
```rbs
class Set[unchecked out E] # & class Array[unchecked out E]
  …
  def inspect: [E < _Inspect] () -> String
  …
end
```